src/summary_statistics/means.rs

```diff
@@ -0,0 +1,106 @@
+use ndarray::{Data, Dimension, ArrayBase};
+use num_traits::{FromPrimitive, Float, Zero};
+use std::result::Result;
+use std::ops::{Add, Div};
+use super::SummaryStatisticsExt;
+
+
+impl<A, S, D> SummaryStatisticsExt<A, S, D> for ArrayBase<S, D>
+where
+    S: Data<Elem = A>,
+    D: Dimension,
+{
+    fn mean(&self) -> Result<A, &'static str>
+    where
+        A: Clone + FromPrimitive + Add<Output=A> + Div<Output=A> + PartialEq + Zero
```

Comment:
What it the `PartialEq` bound used for?

Reply:
Probably it survived some early iterations I was doing. I got rid of it.